chain/chain/src/update_shard.rs

```diff
@@ -147,6 +147,47 @@ pub(crate) fn process_shard_update(
     }
 }
 
+/// Processes shard updates for the execution contexts range which must
+/// correspond to missing chunks for some shard.
+/// `current_chunk_extra` must correspond to `ChunkExtra` just before
+/// execution; in the end it will correspond to the latest execution
+/// result.
+pub(crate) fn process_missing_chunks_range(
+    parent_span: &tracing::Span,
+    current_chunk_extra: &mut ChunkExtra,
```

Comment:
Please avoid mutating arguments when possible.
In this case you clone chunk extra every time anyway:
```
result.push(ChunkExtra{ state_root:  apply_result.new_root, ..current_chunk_extra });
```

Reply:
Makes sense. I think `mut current_chunk_extra: ChunkExtra` is fine here as it is not implicitly returned back